go/libraries/doltcore/sqle/dtables/conflicts_tables_root_objects.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dtables
+
+import (
+	"errors"
+	"io"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+)
+
+// NewConflictsTableRootObject returns a new conflicts table for root objects.
+func NewConflictsTableRootObject(ctx *sql.Context, baseTableName string, rootObject doltdb.ConflictRootObject, root doltdb.RootValue, rs RootSetter) (sql.Table, error) {
+	return ConflictsTableRootObject{
+		baseTableName: baseTableName,
+		sqlSch:        rootObject.Schema(baseTableName),
+		root:          root,
+		rootObject:    rootObject,
+		rs:            rs,
+	}, nil
+}
+
+// ConflictsTableRootObject is a sql.Table implementation that provides access to the conflicts that exist on a root
+// object.
+type ConflictsTableRootObject struct {
```

Comment:
"ConflictsTableRootObject" makes it sound like this is a type of "RootObject" to me. You might consider "ConflictRootObjectTable" to make it sound more like this is a type of table for a ConflictRootObject.